lib/src/emoji_picker_utils.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:emoji_picker_flutter/emoji_picker_flutter.dart';
+import 'package:emoji_picker_flutter/src/recent_emoji.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// Helper class that provides extended usage
+
+class EmojiPickerUtils {
+  static Map<Category, Map<String, String>?>? _availableCategoryEmoji;
+  static final List<Emoji> _allAvailableEmojiEntities = [];
+
+  /// Returns list of recently used emoji from cache
+  static Future<List<RecentEmoji>> getRecentEmojis() async {
+    final prefs = await SharedPreferences.getInstance();
+    var emojiJson = prefs.getString('recent');
+    if (emojiJson == null) {
+      return [];
+    }
+    var json = jsonDecode(emojiJson) as List<dynamic>;
+    return json.map<RecentEmoji>(RecentEmoji.fromJson).toList();
+  }
+
+  /// Returns map of all the available category emojis
+  static Future<Map<Category, Map<String, String>?>?>
+      getAvailableCategoryEmoji() async {
+    if (_availableCategoryEmoji == null) {
+      // Filter recent category emojis
+      final categories =
+          Category.values.where((category) => category != Category.RECENT);
+
+      // Get all available emojis from cached
+      final futures = categories.map((category) async {
+        final emojis = await restoreFilteredEmojis(category.name);
+        return emojis;
+      });
+
+      final allAvailableEmojis = await Future.wait(futures);
+
+      _availableCategoryEmoji =
+          Map.fromIterables(categories, allAvailableEmojis);
+    }
+    return _availableCategoryEmoji;
+  }
```

Comment:
I also considered this problem at the beginning, but I think that since it should be used until the EmojiPicker is used, there may be a problem with my idea.😄

Reply:
I also considered this problem at first, but what I thought was that since this tool class should be used only when the pop-up window is used, there may be a problem with my idea.😄